### PR TITLE
Fix issue #159

### PR DESCRIPTION
**Pull request by cnderrauber**
use timestamp for AudeioLevel observer
change smoothinterval default to 2 for more sensitive

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func NewConfig(confString string, c *cli.Context) (*Config, error) {
 			ActiveLevel:     30, // -30dBov = 0.03
 			MinPercentile:   40,
 			UpdateInterval:  500,
-			SmoothIntervals: 4,
+			SmoothIntervals: 2,
```
Comment (cnderrauber) [inline pkg/config/config.go:149]: Test found that SmoothInterval set to 4 willl cause the speaker state reaction a little slow. I have test set default to 2 will more sensitive. But don't know is there other reason to set SmoothIntervals to 4, if it has, can revert this change

File: pkg/sfu/buffer/buffer.go [hunk 2]
```diff
@@ -382,11 +382,14 @@ func (b *Buffer) calc(pkt []byte, arrivalTime int64) {
 	}
 	b.lastTransit = transit
 
-	if b.audioLevel {
+	if b.audioLevel && headPkt {
```
Comment (boks1971) [inline pkg/sfu/buffer/buffer.go:385]: In practice this is not going to be a problem as audio packets are small.

But, to ensure protocol compliance, I think we should remove the `headPkt` check here. Assume that an audio frame is split into two RTP packets and only the first one (the one which is not `headPkt`) has the ssrc-audio-level extension and the second one (one with `headPkt`) does not have it. Then this code will not get the ssrc-audio-level. There is already a check below for 0 duration. That should take care of the case where a packet has been split into two and both have ssrc-audio-level as both those should have same RTP timestamp. That will take care of avoiding double counting.

File: pkg/rtc/audiolevel.go [hunk 3]
```diff
@@ -7,57 +7,57 @@ import (
 
 const (
 	// number of audio frames for observe window
-	observeFrames    = 25 // webrtc default opus frame size 20ms, 25*20=500ms matching default UpdateInterval
+	observeDuration  = 500 // 500ms matching default UpdateInterval
```
Comment (boks1971) [inline pkg/rtc/audiolevel.go:10]: can leave out `matching default UpdateInterval` as that is not relevant in this time based approach.

File: pkg/rtc/audiolevel.go [hunk 4]
```diff
@@ -7,57 +7,57 @@ import (
...
 		}
 	}
 
-	if l.numFrames >= observeFrames {
+	if l.observedDuration >= observeDuration {
 		// compute and reset
-		if l.activeFrames >= l.minActiveFrames {
-			const invObserveFrames = 1.0 / observeFrames
-			level := uint32(l.observeLevel) - uint32(20*math.Log10(float64(l.activeFrames)*invObserveFrames))
+		if l.activeDuration >= l.minActiveDuration {
```
Comment (boks1971) [inline pkg/rtc/audiolevel.go:51]: A few things to note
1. Can remove the `inv...` variable and just divide in the equation. It is trying to find percentage of active time and that inv and multiply confuses me heavily :-)
2. With the move to duration based, it would be good to take the actual duration while calculating percentage activity. For example, you could be getting packets continuously for 400 ms and then DTX kicks in and the next packet comes after another 400 ms, the duration will be 800 ms which is way more than the constant observe duration.
3. Continuing on the above point, if we want to measure every 1/2 second, we should ideally be running a goroutine and update periodically. If we were to do this, then we do not have worry about the above example as much.

File: pkg/sfu/buffer/buffer.go (hunk 2 above)
Comment (cnderrauber) [reply pkg/sfu/buffer/buffer.go:385]: Never see a audio frame split to 2 packets, and just first one has audio level. Does it exist only in theory?

File: pkg/rtc/audiolevel.go (hunk 4 above)
Comment (cnderrauber) [reply pkg/rtc/audiolevel.go:51]: Make sense, but do we need to be so precise  here? We can observe and improve it if some people complain with the speech indicator.

Comment (boks1971) [reply pkg/rtc/audiolevel.go:51]: Yeah, don't need to be precise. We can always wait and watch. But, it is simple enough to make this change now rather than wait. So, thought of suggesting it.

**Review by boks1971 — APPROVED** [review]

File: pkg/sfu/buffer/buffer.go (hunk 2 above)
Comment (boks1971) [reply pkg/sfu/buffer/buffer.go]: Only in theory Jie. I have not seen it in practice. But, just in case some crazy implementation does it, let's not get bitten by it :-)
